tests/integration/test_fim/test_files/test_ambiguous_confs/test_ignore_works_over_restrict.py

```diff
@@ -164,35 +163,41 @@ def test_ignore_works_over_restrict(folder, filename, triggers_event, tags_to_ap
 
     # Create file that must be ignored
     logger.info(f'Adding file {os.path.join(testdir1, filename)}, content: ""')
     create_file(REGULAR, folder, filename, content='')
 
-    # Go ahead in time to let syscheck perform a new scan if mode is scheduled
-    logger.info(f'Time travel: {scheduled}')
-    check_time_travel(scheduled, monitor=wazuh_log_monitor)
+    # Time that the test will wait for the event or log to be generated.
+    logs_generation_timeout = 5  # seconds
+    logger.info(f'Waiting up to {logs_generation_timeout} seconds for FIM to generate the log')
 
     if triggers_event:
         logger.info('Checking the event...')
-        event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+        event = wazuh_log_monitor.start(timeout=logs_generation_timeout,
                                         callback=callback_detect_event,
                                         error_message=f'Did not receive expected "Sending FIM event" '
                                                       f'event for file {os.path.join(testdir1, filename)}').result()
 
         assert event['data']['type'] == 'added', 'Event type not equal'
         assert event['data']['path'] == os.path.join(folder, filename), 'Event path not equal'
     else:
-        while True:
-            ignored_file = wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_ignore,
-                                                   error_message=f'Did not receive expected '
-                                                                 f'"Ignoring ... due to ..." event for file '
-                                                                 f'{os.path.join(testdir1, filename)}').result()
-
-            if ignored_file == os.path.join(folder, filename):
-                break
+        regex = CB_IGNORING_DUE_TO_PATTERN if 'valid_no_regex' in tags_to_apply else CB_IGNORING_DUE_TO_SREGEX
+        logger.info('Checking the logs...')
+
+        ignored_file = wazuh_log_monitor.start(timeout=logs_generation_timeout,
+                                               callback=generate_monitoring_callback(regex),
+                                               error_message=f'Did not receive expected '
+                                                             f'"Ignoring ... due to ..." event for file '
+                                                             f'{os.path.join(testdir1, filename)}').result()
+
+        if ignored_file == os.path.join(folder, filename):
+            # The ignored file is the same as the one we are looking for.
+            return
+
+        # In some ocations ignored_file is the previous file that was ignored, so we check again.
+        test_ignore_works_over_restrict(folder, filename, triggers_event, tags_to_apply, get_configuration,
+                                        configure_environment, restart_syscheckd, wait_for_fim_start)
```

Comment:
Great, the test was updated to use accumulative responses and assert into that instead of using recursive test function calls in commit [5592473](https://github.com/wazuh/wazuh-qa/pull/3078/commits/55924731e03129bce9beeb0f88b49dc1a9fbf9f9)